commodore/dependency_mgmt.py

```diff
@@ -52,13 +74,48 @@ def _discover_components(cfg) -> Tuple[List[str], Dict[str, str]]:
                 msg += f" aliased to {alias}"
             click.echo(msg)
         components.add(cn)
+        all_component_aliases.setdefault(alias, set()).add(cn)
+
+    component_aliases: Dict[str, str] = {}
 
-        if alias in component_aliases:
-            pc = component_aliases[alias]
+    for alias, cns in all_component_aliases.items():
+        if len(cns) == 0:
+            # NOTE(sg): This should never happen, but we add it for completeness' sake.
+            raise ValueError(
+                f"Discovered component alias '{alias}' with no associated components"
+            )
+
+        if len(cns) > 1:
+            if alias in cns:
+                other_aliases = list(cns - set([alias]))
+                if len(other_aliases) > 1:
+                    clist = _format_component_list(other_aliases)
+                    raise KeyError(
+                        f"Components {clist} alias existing component '{alias}'"
+                    )
+
+                # If this assertion fails we have a problem, since `other_aliases` is
+                # the result of removing a single element from a set which contains
+                # multiple elements and we've already handled the case for len() > 1.
+                # Since we don't mind if it's optimized out in some cases, we annotate
+                # it with `nosec` so bandit doesn't complain about it.
+                assert len(other_aliases) == 1  # nosec
+                raise KeyError(
+                    f"Component '{other_aliases[0]}' "
+                    + f"aliases existing component '{alias}'"
+                )
```

Comment:
I was confused by this  for a second. I would personally probably check above that we do not add the component to `all_component_aliases` if `alias == cn`

Reply:
Ah wait no I misunderstood. Ignore that